tests/phpunit/elementor/core/experiments/test-manager.php

```diff
@@ -372,36 +390,101 @@ public function test_get_active_features() {
 	}
 
 	public function test_is_feature_active() {
-		$this->add_test_feature();
+		$this->add_test_feature( [
+			'name' => 'test_feature',
+			'default' => Experiments_Manager::STATE_ACTIVE,
+		] );
 
 		$is_test_feature_active = $this->experiments->is_feature_active( 'test_feature' );
 
 		$this->assertTrue( $is_test_feature_active );
 	}
 
 	public function test_is_feature_active__new_site() {
-		update_option( Manager::get_install_history_meta(), [
-			time() => '3.1.0',
+		update_option( Manager::INSTALLS_HISTORY_META, [
+			'3.1.0' => time(),
 		] );
 
-		$this->add_test_feature( [
+		$this->experiments->add_feature( [
+			'name' => 'test_feature',
 			'default' => Experiments_Manager::STATE_INACTIVE,
+			'mutable' => false,
 			'new_site' => [
 				'default_active' => true,
 				'minimum_installation_version' => '3.1.0',
 			],
 		] );
 
-		$experiments = $this->experiments;
+		$is_test_feature_active = $this->experiments->is_feature_active( 'test_feature' );
 
-		$is_test_feature_active = $experiments->is_feature_active( 'test_feature' );
+		$this->assertTrue( $is_test_feature_active );
+	}
+
+
+	public function test_is_feature_active__new_site_should_be_active() {
+		update_option( Manager::INSTALLS_HISTORY_META, [
+			'3.1.0' => time(),
+		] );
+
+		$this->experiments->add_feature( [
+			'name' => 'test_feature',
+			'default' => Experiments_Manager::STATE_INACTIVE,
+			'mutable' => false,
+			'new_site' => [
+				'default_active' => true,
+				'minimum_installation_version' => '3.0.0',
+			],
+		] );
+
+		$is_test_feature_active = $this->experiments->is_feature_active( 'test_feature' );
+
+		$this->assertTrue( $is_test_feature_active );
+	}
+
+	public function test_is_feature_active__new_site_should_be_inactive() {
+		update_option( Manager::INSTALLS_HISTORY_META, [
+			'3.0.0' => time(),
+		] );
+
+		$this->experiments->add_feature( [
+			'name' => 'test_feature',
+			'default' => Experiments_Manager::STATE_INACTIVE,
+			'mutable' => false,
+			'new_site' => [
+				'default_active' => true,
+				'minimum_installation_version' => '3.1.0',
+			],
+		] );
+
+		$is_test_feature_active = $this->experiments->is_feature_active( 'test_feature' );
+
+		$this->assertFalse( $is_test_feature_active );
+	}
+
+	public function test_is_feature_active__new_site_with_beta() {
+		update_option( Manager::INSTALLS_HISTORY_META, [
+			'3.1.0-beta' => time(),
+			'3.1.0-beta2' => time() + 1,
+		] );
+
+		$this->experiments->add_feature( [
+			'name' => 'test_feature',
+			'default' => Experiments_Manager::STATE_INACTIVE,
+			'mutable' => false,
+			'new_site' => [
+				'default_active' => true,
+				'minimum_installation_version' => '3.1.0',
+			],
+		] );
+
+		$is_test_feature_active = $this->experiments->is_feature_active( 'test_feature' );
```

Comment:
maybe also one for inactive?
and maybe just use a data provider for all those tests

Reply:
OK